### PR TITLE
Add logging to Must Gather

### PR DESCRIPTION
**Pull request by nilsanderselde**
Adds logging to Must Gather Geneva Action and changes kubeaction instantiation to use a factory as suggested in [the original PR](https://github.com/Azure/ARO-RP/pull/470).

### CODE REVIEW

File: pkg/frontend/admin_openshiftcluster_kubernetesobjects_get.go [hunk 1]
```diff
@@ -24,12 +25,12 @@ func (f *frontend) getAdminKubernetesObjects(w http.ResponseWriter, r *http.Requ
 	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
 	r.URL.Path = filepath.Dir(r.URL.Path)
 
-	b, err := f._getAdminKubernetesObjects(ctx, r)
+	b, err := f._getAdminKubernetesObjects(ctx, r, kubeactions.New(log, f.env))
```
Comment (asalkeld) [inline pkg/frontend/admin_openshiftcluster_kubernetesobjects_get.go:28]: why pass this through when you can do the same in the function below? Not a biggie, just seems unnecessary.

File: pkg/frontend/admin_openshiftcluster_kubernetesobjects_get.go [hunk 2]
```diff
@@ -50,9 +51,9 @@ func (f *frontend) _getAdminKubernetesObjects(ctx context.Context, r *http.Reque
 	}
 
 	if name != "" {
-		return f.kubeActions.Get(ctx, doc.OpenShiftCluster, kind, namespace, name)
+		return ka.Get(ctx, doc.OpenShiftCluster, kind, namespace, name)
```
Comment (asalkeld) [inline pkg/frontend/admin_openshiftcluster_kubernetesobjects_get.go:54]: Are the unittests working? What is your plan with mocking?

Comment (nilsanderselde) [reply pkg/frontend/admin_openshiftcluster_kubernetesobjects_get.go:54]: They aren't at the moment. I will work on that.

File: pkg/frontend/admin_openshiftcluster_kubernetesobjects_get.go (hunk 1 above)
Comment (nilsanderselde) [reply pkg/frontend/admin_openshiftcluster_kubernetesobjects_get.go:28]: Yeah, I think you're right.

**Comment by nilsanderselde** [comment]
Need to rethink this, due to the issue with mocking kubeactions. Reopening https://github.com/Azure/ARO-RP/pull/470 for now.

File: pkg/frontend/frontend.go [hunk 3]
```diff
@@ -50,7 +49,6 @@ type frontend struct {
 	cipher  encryption.Cipher
 
 	ocEnricher             clusterdata.OpenShiftClusterEnricher
-	kubeActions            kubeactions.Interface
```
Comment (jim-minter) [inline pkg/frontend/frontend.go:53]: `kubeActionsFactory func(*logrus.Entry, env.Interface) kubeactions.Interface`

File: pkg/frontend/admin_openshiftcluster_kubernetesobjects_get.go (hunk 1 above)
Comment (jim-minter) [reply pkg/frontend/admin_openshiftcluster_kubernetesobjects_get.go:28]: pass log through to the child function

File: pkg/frontend/admin_openshiftcluster_kubernetesobjects_get.go [hunk 4]
```diff
@@ -24,12 +25,12 @@ func (f *frontend) getAdminKubernetesObjects(w http.ResponseWriter, r *http.Requ
...
 
-	b, err := f._getAdminKubernetesObjects(ctx, r)
+	b, err := f._getAdminKubernetesObjects(ctx, r, kubeactions.New(log, f.env))
 
 	adminReply(log, w, nil, b, err)
 }
 
-func (f *frontend) _getAdminKubernetesObjects(ctx context.Context, r *http.Request) ([]byte, error) {
+func (f *frontend) _getAdminKubernetesObjects(ctx context.Context, r *http.Request, ka kubeactions.Interface) ([]byte, error) {
 	vars := mux.Vars(r)
```
Comment (jim-minter) [inline pkg/frontend/admin_openshiftcluster_kubernetesobjects_get.go:34]: ka := f.kubeActionsFactory(log, f.env)

File: pkg/frontend/frontend.go [hunk 5]
```diff
@@ -67,15 +65,14 @@ type Runnable interface {
 }
 
 // NewFrontend returns a new runnable frontend
-func NewFrontend(ctx context.Context, baseLog *logrus.Entry, _env env.Interface, db *database.Database, apis map[string]*api.Version, m metrics.Interface, cipher encryption.Cipher, kubeActions kubeactions.Interface, resourcesClientFactory resourcesClientFactory) (Runnable, error) {
+func NewFrontend(ctx context.Context, baseLog *logrus.Entry, _env env.Interface, db *database.Database, apis map[string]*api.Version, m metrics.Interface, cipher encryption.Cipher, resourcesClientFactory resourcesClientFactory) (Runnable, error) {
```
Comment (jim-minter) [inline pkg/frontend/frontend.go:68]: receive kubeActionsFamily here.

In non-test set kubeActionsFactory to kubeactions.New. In test we'll override it to some mock function.

**Comment by jim-minter** [comment]
this work may seem somewhat futile or overblown but it is also preparatory work to putting the k8s clients into the kubeactions struct, which will enable it to be unit tested properly.

**Pull request merged**